lib/config.js

```diff
@@ -229,6 +231,16 @@ function getLocalConfig(thisConfig, directory) {
             continue;
         }
 
+        // Check for root flag
+        if (localConfig.root === true) {
+            rootPath = path.dirname(localConfigFile);
+        }
+
+        // If root flag is set, don't consider file if it is above root
+        if (rootPath && !pathIsInside(path.dirname(localConfigFile), rootPath)) {
```

Comment:
Okay, I had an idea that might simplify all of this. What if we said that `root` specified the root config instead of the root directory? That way, we could just `break` out of this loop and everything is nice and simple. Thoughts?